buildenv

```diff
@@ -175,49 +160,122 @@ zopen_build() {
+  git apply ${ZOPEN_ROOT}/patches/third_party/fuzztest/src/git.diff
+  echo "  -- googletest"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/googletest/src
+  git apply ${ZOPEN_ROOT}/patches/third_party/googletest/src/git.diff
+  echo "  -- protobuf_chrome"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/protobuf_chrome
+  git apply ${ZOPEN_ROOT}/patches/third_party/protobuf_chrome/git.diff
+  echo "  -- re2"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/re2/src
+  git apply ${ZOPEN_ROOT}/patches/third_party/re2/src/git.diff
+  echo "  -- zlib"
+  cd ${ZOPEN_ROOT}/v8base/v8/third_party/zlib
+  git apply ${ZOPEN_ROOT}/patches/third_party/zlib/git.diff
+  echo "Apply patches for overall repo complete"
+  echo "Apply patches for overall repo complete"
+  echo "Apply patches for overall repo complete"
+
+  echo "Perform file overlays"
+  echo "Perform file overlays"
+  echo "Perform file overlays"
+  cd ${ZOPEN_ROOT}/v8base/v8/
+  echo "nothing done. clarify.  TODO"
+  echo "  -- patches/buildtools/zos/ar"
+  echo "  -- patches/src/base/debug/stack_trace_zos.cc"
+  echo "  -- patches/src/base/platform/platform-zos.cc"
+  echo "  -- patches/src/heap/src/asm/zos/push_registers_asm.cc"
+  echo "  -- patches/src/snapshot/embedded/platform-embedded-file-writer-zos.cc"
+  echo "  -- patches/src/snapshot/embedded/platform-embedded-file-writer-zos.h"
+  echo "Perform file overlays complete"
+  echo "Perform file overlays complete"
+  echo "Perform file overlays complete"
 
   echo "Apply pax files to buildtools"
-  pushd buildtools
+  echo "Apply pax files to buildtools"
+  echo "Apply pax files to buildtools"
+  cd ${ZOPEN_ROOT}/v8base/v8/buildtools
   pax -p p -rzf ${ZOPEN_ROOT}/patches/buildtools/buildtools.zos.20231122.92b79f4d75.pax
-  popd
+  echo "Apply pax files to buildtools complete"
+  echo "Apply pax files to buildtools complete"
+  echo "Apply pax files to buildtools complete"
+
+  echo "** APPLY PATCHES, PAX and OVERLAYS COMPLETE ***********" 
+  echo "** APPLY PATCHES, PAX and OVERLAYS COMPLETE ***********" 
+  echo "** APPLY PATCHES, PAX and OVERLAYS COMPLETE ***********" 
+
+
   #
   # Prepare v8 for building
   #
-  pushd buildtools/zos
+  cd ${ZOPEN_ROOT}/v8base/v8/buildtools/zos
   echo "change dir to v8base/v8/buildtools/zos level..."
   echo "make link to gn in v8base/v8/buildtools/zos"
   echo "using GN_HOME: ${GN_HOME}"
   echo "cwd: `pwd`"
   ln -s "${GN_HOME}/gn" .
-  popd
+  cd ../..
   export PATH="${PWD}:${ZOPEN_ROOT}/v8base/v8/buildtools/zos:$PATH" # for the 'ar' wrapper
   echo "ZOPEN_ROOT : ${ZOPEN_ROOT}"
   echo "PATH : ${PATH}"
-  #JFD EARLY EXIT
-  #echo "JFD EARLY EXIT"
-  #exit
+
+
 
   #
   # Do the v8 build
   #
   echo "Generate ninja files using gn..."
   gn -v gen out/zos_s390x.release --args="is_debug=false treat_warnings_as_errors=false ${COMPONENT_BUILD}"
+  # Why was this done later?  Should we not do it now once the build output dir is there 
+  # and when the build happens?
```

Comment:
This was a bug, and should explain why libicuuc.so couldn't be found in your component build, unless there was an error  prior to it being created.

Reply:
ok thanks